extensions/functions_arithmetic.yaml

```diff
@@ -742,6 +742,52 @@ aggregate_functions:
           - value: fp64
         nullability: DECLARED_OUTPUT
         return: fp64?
+  - name: "corr"
+    description: Calculates correlation of two set of values.
+    impls:
+      - args:
+          - name: rounding
+            options: [ TIE_TO_EVEN, TIE_AWAY_FROM_ZERO, TRUNCATE, CEILING, FLOOR ]
+            required: false
+          - value: fp32
+          - value: fp32
+        nullability: DECLARED_OUTPUT
+        return: fp32?
+      - args:
+          - name: rounding
+            options: [ TIE_TO_EVEN, TIE_AWAY_FROM_ZERO, TRUNCATE, CEILING, FLOOR ]
+            required: false
+          - value: fp64
+          - value: fp64
+        nullability: DECLARED_OUTPUT
+        return: fp64?
+  - name: "mode"
+    description: Calculates mode for a set of values.
```

Comment:
I feel like there should be some description of what should happen if the mode is not uniquely defined, for example for [1, 1, 2, 2]. Does that yield 1, 2, either 1 or 2 at the discretion of the consumer, or null?

The mode of a set of floating-point values is also weird, since mode requires a strict equality operator that languages usually shy away from defining for floats, though Substrait doesn't generally consider floats to be special and implicitly assumes an equality operator exists for all types (since there are no specified type constraints on set operations for example). Flipping that around actually, since mode *only* requires an equality operator, shouldn't the mode be defined for all types rather than only on simple numeric types?

Reply:
What do you mean by simple numeric types?

Reply:
[Simple](https://substrait.io/types/type_classes/#simple-types) types representing a number, i.e. exactly the types you have now exclusively defined `mode` for. I'm saying the logic of `mode` does not depend on whether a type represents a number or not, so why not just define it for *all* types? For example, the mode of the strings `["a", "b", "c", "b"]` is `"b"`, because that value appears more often than the other values.

Reply:
so basically we could use `any` here right?

Reply:
I think so. It might be a bit weird though nonetheless. Maybe we should just leave it as is for now.